homeassistant/components/motioneye/camera.py

```diff
@@ -0,0 +1,206 @@
+
+    @callback
+    def camera_add(camera: dict[str, Any]) -> None:
+        """Add a new motionEye camera."""
+        async_add_entities(
+            [
+                MotionEyeMjpegCamera(
+                    entry.entry_id,
+                    entry.data.get(
+                        CONF_SURVEILLANCE_USERNAME, DEFAULT_SURVEILLANCE_USERNAME
+                    ),
+                    entry.data.get(CONF_SURVEILLANCE_PASSWORD, ""),
+                    camera,
+                    entry_data[CONF_CLIENT],
+                    entry_data[CONF_COORDINATOR],
+                )
+            ]
+        )
+
+    listen_for_new_cameras(hass, entry, camera_add)
+    return True
+
+
+class MotionEyeMjpegCamera(MjpegCamera, CoordinatorEntity):
+    """motionEye mjpeg camera."""
+
+    def __init__(
+        self,
+        config_entry_id: str,
+        username: str,
+        password: str,
+        camera: dict[str, Any],
+        client: MotionEyeClient,
+        coordinator: DataUpdateCoordinator,
+    ):
+        """Initialize a MJPEG camera."""
+        self._surveillance_username = username
+        self._surveillance_password = password
+        self._client = client
+        self._camera_id = camera[KEY_ID]
+        self._device_id = get_motioneye_device_unique_id(
+            config_entry_id, self._camera_id
+        )
+        self._unique_id = get_motioneye_entity_unique_id(
+            config_entry_id, self._camera_id, TYPE_MOTIONEYE_MJPEG_CAMERA
+        )
+        self._motion_detection_enabled: bool = camera.get(KEY_MOTION_DETECTION, False)
+        self._available = MotionEyeMjpegCamera._is_acceptable_streaming_camera(camera)
+
+        # motionEye cameras are always streaming or unavailable.
+        self.is_streaming = True
+
+        MjpegCamera.__init__(
+            self,
+            {
+                CONF_VERIFY_SSL: False,
+                **self._get_mjpeg_camera_properties_for_camera(camera),
+            },
+        )
+        CoordinatorEntity.__init__(self, coordinator)
+
+    def _get_mjpeg_camera_properties_for_camera(
+        self, camera: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Convert a motionEye camera to MjpegCamera internal properties."""
+        auth = None
+        if camera.get(KEY_STREAMING_AUTH_MODE) in [
+            HTTP_BASIC_AUTHENTICATION,
+            HTTP_DIGEST_AUTHENTICATION,
+        ]:
+            auth = camera[KEY_STREAMING_AUTH_MODE]
+
+        return {
+            CONF_NAME: camera[KEY_NAME],
+            CONF_USERNAME: self._surveillance_username if auth is not None else None,
+            CONF_PASSWORD: self._surveillance_password if auth is not None else None,
+            CONF_MJPEG_URL: self._client.get_camera_stream_url(camera) or "",
+            CONF_STILL_IMAGE_URL: self._client.get_camera_snapshot_url(camera),
+            CONF_AUTHENTICATION: auth,
+        }
+
+    def _set_mjpeg_camera_state_for_camera(self, camera: dict[str, Any]) -> None:
```

Comment:
For things that are safe to run inside the event loop, please decorate with `@callback`. 
```suggestion
    @callback
    def _set_mjpeg_camera_state_for_camera(self, camera: dict[str, Any]) -> None:
```

Reply:
Done. For my understanding for future, given that this function isn't called directly from the event_loop (but rather indirectly from `_handle_coordinator_update`, which _is_ decorated) -- why is @callback appropriate/required here?

Reply:
It makes it easy for the reviewer to know how the function is intended to be used. If I see a non-annotated function, I assume it's meant as a sync function and is allowed to do I/O. If it's async or `@callback`, it makes it easier to spot functions that do I/O .